python/python/tests/test_async_client.py

```diff
@@ -7462,6 +7462,90 @@ async def test_cluster_flushdb(self, redis_client: GlideClusterClient):
             assert await redis_client.flushdb(FlushMode.SYNC, AllPrimaries()) is OK
             assert await redis_client.dbsize() == 0
 
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_sscan(self, redis_client: GlideClusterClient):
+        key1 = f"{{key}}-1{get_random_string(5)}"
+        key2 = f"{{key}}-2{get_random_string(5)}"
+        initial_cursor = "0"
+        result_cursor_index = 0
+        result_collection_index = 1
+        default_count = 10
+        num_members = list(
+            map(str, range(50000))
+        )  # Use large dataset to force an iterative cursor.
+        char_members = ["a", "b", "c", "d", "e"]
+
+        # Empty set
+        result = await redis_client.sscan(key1, initial_cursor)
+        assert result[result_cursor_index] == initial_cursor
+        assert result[result_collection_index] == []
+
+        # Negative cursor
+        result = await redis_client.sscan(key1, "-1")
+        assert result[result_cursor_index] == initial_cursor
+        assert result[result_collection_index] == []
+
+        # Result contains the whole set
+        assert await redis_client.sadd(key1, char_members) == len(char_members)
+        result = await redis_client.sscan(key1, initial_cursor)
+        assert result[result_cursor_index] == initial_cursor
+        assert len(result[result_collection_index]) == len(char_members)
+        assert set(result[result_collection_index]).issubset(set(char_members))
+
+        result = await redis_client.sscan(key1, initial_cursor, match="a")
+        assert result[result_cursor_index] == initial_cursor
+        assert set(result[result_collection_index]).issubset(set(["a"]))
+
+        # Result contains a subset of the key
+        assert await redis_client.sadd(key1, num_members) == len(num_members)
+        result_cursor = "0"
+        result_values = set()  # type: set[str]
+        result = await redis_client.sscan(key1, result_cursor)
+        result_cursor = str(result[result_cursor_index])
```

Comment:
Nit: We shouldn't need to call str() for cursors because they should already be strings.

Reply:
`mypy` complains in CI if I don't cast. See error:
```
../python/python/tests/test_async_client.py:7563: error: Incompatible types in assignment (expression has type "Union[str, List[str]]", variable has type "str")  [assignment]
../python/python/tests/test_async_client.py:7576: error: Incompatible types in assignment (expression has type "Union[str, List[str]]", variable has type "str")  [assignment]